### PR TITLE
add size property into Image object containing width and height of each image (thumbnail, big).

### PR DESCRIPTION
**Pull request by franek**
Can be useful for some themes to disinguish orientation of the image.

### CODE REVIEW

**Comment by saimn** [comment]
It would be a good thing to add this, thanks for proposing it. However I would like a different api: `get_size` should work on only one file (not specific case for the thumbnail, which is just another file), and there should be two properties, `Image.size` and `Image.thumb_size` for the thumbnail.
Actually I'm not sure if the `get_size` function is needed, but one advantage of putting this in a function should be to have a unit test, it would great if you can add one ;).

File: sigal/gallery.py [hunk 1]
```diff
@@ -371,7 +379,7 @@ def thumbnail(self):
                     except:
                         self.logger.error("Failed to open %s", f.src_path)
                     else:
-                        if im.size[0] > im.size[1]:
+                        if im.size[0] > im.size[1]:
```
Comment (saimn) [inline sigal/gallery.py]: In this block of code there is a call to `PILImage.open` which could be simplified by using the new property, if you want to change that.

Comment (franek) [inline sigal/gallery.py]: I have tried but I am a python "hacker" (bidouilleur) and did not succeed (got some errors with tests). Sorry about that.

File: sigal/image.py [hunk 2]
```diff
@@ -160,6 +161,20 @@ def process_image(filepath, outpath, settings):
     return Status.SUCCESS
 
 
+def get_size(file_path):
+    """Return image size (width and height)."""
+    logger = logging.getLogger(__name__)
+    try:
+        im = PILImage.open(file_path)
+    except:
+        logger.error("Failed to open %s", file_path)
```
Comment (saimn) [inline sigal/image.py]: With this try/except, the try/except in the properties is useless and can be removed. Also the error message could be more specific, like above (`Could not read size ...`), and specifying the list of exceptions (`IOError, IndexError, TypeError, AttributeError` should be fine) is better practise.

File: tests/test_image.py [hunk 3]
```diff
@@ -115,3 +115,13 @@ def test_exif_gps(tmpdir):
...
+
+def test_get_size(tmpdir):
+    """Test reading out image size"""
+
+    test_image = 'flickr_jerquiaga_2394751088_cc-by-nc.jpg'
+    src_file = os.path.join(CURRENT_DIR, 'sample', 'pictures', 'dir1', 'test1',
+                            test_image)
+
+    result = get_size(src_file)
+    assert result == {'height': 800, 'width': 600}
```
Comment (saimn) [inline tests/test_image.py]: Please add a test for a missing file to test the except clause.

File: sigal/gallery.py (hunk 1 above)
Comment (saimn) [inline sigal/gallery.py]: Ok, no problem, I will take a look.

**Comment by saimn** [comment]
Thanks François !

**Pull request merged**